### PR TITLE
[Outreachy Task Submission] Make Icon-Only Buttons and Clickable Cards Accessible By Screen Readers

### PR DESCRIPTION
**Pull request by Aquilaafuadajo**
This pr aims to address [4756](https://github.com/ushahidi/platform/issues/4756).

**Description**
Currently some icon-only buttons and clickable cards on the Ushahidi Platform are not accessible by screen readers, so when a user tabs to that button or card location the screen reader just announces "button". This behaviour can confuse a user that relies heavily on keyboard for navigation as they won't know what the button or card does. I have attached a screen recording of the issue for better understanding.

**Before Fix**

https://github.com/ushahidi/platform-client-mzima/assets/46582086/2028f523-46ec-45a7-bb73-f509e0893d2b

**After Fix**

https://github.com/ushahidi/platform-client-mzima/assets/46582086/e25882f2-146f-41cc-9f99-827395a4f1a1

**How To Test**
1. Install [google chrome screen reader extension](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn)
2. Turn on the screen reader extension
3. Navigate through the Ushahidi data view page by pressing the tab key on your keyboard.
4. Observe that it pronounces the right button actions when the icon buttons receive focus.

### CODE REVIEW

**Comment by Angamanga** [comment]
Thank you for your PR @Aquilaafuadajo  🙏. Could you please add a short description and a checklist for testing the fix as well and then we'll review it 👍

**Comment by Aquilaafuadajo** [comment]
@Angamanga Thanks for feedback. I have done as you asked 👍🏽

File: apps/web-mzima-client/src/app/post/post-head/post-head.component.html [hunk 1]
```diff
@@ -2,10 +2,11 @@
   <mzima-client-button
     fill="clear"
     color="gray"
-    tabindex="-1"
+    ariaLabel="Edit this post"
     *ngIf="editable"
     [iconOnly]="true"
     (buttonClick)="editPost($event)"
+    (keydown.enter)="editPost($event)"
     class="post-head__control"
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/post/post-head/post-head.component.html:10]: Can you explain a little more about this change?

File: apps/web-mzima-client/src/app/post/post-head/post-head.component.html [hunk 2]
```diff
@@ -25,9 +27,10 @@
     *ngIf="!!user?.userId"
     fill="clear"
     color="gray"
-    tabindex="-1"
+    ariaLabel="More options"
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/post/post-head/post-head.component.html:30]: Is there a reason why you removed the tabindex here?

File: apps/web-mzima-client/src/app/post/post-preview/post-preview.component.html [hunk 3]
```diff
@@ -31,7 +31,7 @@
 
   <div class="post__main">
     <div class="post__content">
-      <h3 *ngIf="post.title?.length || post.content?.length">{{ post.title || post.content }}</h3>
+      <p *ngIf="post.title?.length || post.content?.length">{{ post.title || post.content }}</p>
       <p *ngIf="post.title?.length && post.content?.length">{{ post.content }}</p>
```
Comment (Angamanga) [inline apps/web-mzima-client/src/app/post/post-preview/post-preview.component.html]: Can you explain a little more why the h3 is changed to a p here?

**Review by Angamanga — CHANGES_REQUESTED** [review]
Thank you for your PR, great finding and start of the work! Please see my questions about clarifications in the code 👍

File: apps/web-mzima-client/src/app/post/post-head/post-head.component.html (hunk 1 above)
Comment (Aquilaafuadajo) [reply apps/web-mzima-client/src/app/post/post-head/post-head.component.html:10]: @Angamanga this line here listens for keyboard enter event on the edit icon button when navigating with the keyboard using tab.

File: apps/web-mzima-client/src/app/post/post-head/post-head.component.html (hunk 2 above)
Comment (Aquilaafuadajo) [reply apps/web-mzima-client/src/app/post/post-head/post-head.component.html:30]: tabindex="-1" here prevented the more options icon-button from receiving focus when navigating using only keyboard. Having it here will prevent keyboard users from ever reaching the more options menu for each post cards

File: apps/web-mzima-client/src/app/post/post-preview/post-preview.component.html (hunk 3 above)
Comment (Aquilaafuadajo) [reply apps/web-mzima-client/src/app/post/post-preview/post-preview.component.html]: This was a mistake, the change is meant to be for another file in another pr. Would revert now.

**Pull request closed without merging**